File: Python/7py 3/remove_3.py
"""
|--------------------------------------------------------------------------
| Exclamation marks series #3: Remove all exclamation marks from sentence except at the end
|--------------------------------------------------------------------------
|
"""

from re import sub
def remove(s):
   return sub(r'!+(?!!*$)', '', s)


def remove(s):
   num = 0
   for letter in s[::-1]:
      if letter != "!":
         break
      num += 1
   return s.replace("!", "") + "!" * num


def remove(s):
   return s.replace('!', '')+ '!'*(len(s)- len(s.rstrip('!')))


print(remove('Hi!'), 'Hi!')
print(remove('Hi!!!'),'Hi!!!')
print(remove('!Hi'), 'Hi')
print(remove('!Hi!'), 'Hi!')
print(remove('Hi! Hi!'), 'Hi Hi!')
print(remove('Hi'), 'Hi')

File: JavaScript/7js 3/remove_3.js
/*
|--------------------------------------------------------------------------
| Exclamation marks series #3: Remove all exclamation marks from sentence except at the end
|--------------------------------------------------------------------------
|
*/

function remove(string) {
   return string.replace(/!+(?!!*$)/g, "");
}


function remove(str) {
   let last = str.length - 1;
   while (str[last] === "!") last--;
   return [...str].filter((e) => e !== "!").join("") + str.slice(last + 1);
}


function remove(s) {
   return s.replace(/!+([^!])/g, "$1");
}


console.log(remove("Hi!"), "Hi!");
console.log(remove("Hi!!!"), "Hi!!!");
console.log(remove("!Hi"), "Hi");
console.log(remove("!Hi!"), "Hi!");
console.log(remove("Hi! Hi!"), "Hi Hi!");
console.log(remove("Hi"), "Hi");